derp/derphttp/derphttp_client.go

```diff
@@ -648,6 +657,14 @@ const dialNodeTimeout = 1500 * time.Millisecond
 // TODO(bradfitz): longer if no options remain perhaps? ...  Or longer
 // overall but have dialRegion start overlapping races?
 func (c *Client) dialNode(ctx context.Context, n *tailcfg.DERPNode) (net.Conn, error) {
+	if c.nodeDialer != nil {
+		conn, err := c.nodeDialer(ctx, n)
+		// If both conn and err are nil, fallback to default behavior.
+		if conn != nil || err != nil {
```

Comment:
Doesn't this mean we never return conn since we return nil, err below?

Reply:
Ahh, oops! I intended to return both.